lib/active_record_shards/patches-4-2.rb

```diff
@@ -11,3 +11,5 @@
 ActiveRecord::Associations::Builder::HasAndBelongsToMany.include(ActiveRecordShards::DefaultSlavePatches::Rails41HasAndBelongsToManyBuilderExtension)
 
 ActiveRecord::SchemaDumper.prepend(ActiveRecordShards::SchemaDumperExtension)
+
+require 'active_record_shards/association_scope-4-0'
```

Comment:
The patch applies to 4.0, 4.1 and 4.2. Can we just call it active_record_shards/association_scope.rb?

Reply:
I reproduced the pattern used in `connection_switcher-4-0`. That said, we do not support Rails 4.0 and 4.1 so I can just rename the file `association_scope-4-2`